src/movies.js

```diff
@@ -1,16 +1,116 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+    const getAllDirectors = moviesArray => moviesArray.map(elm => elm.director);
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(moviesArray) {
+
+    const movies = moviesArray.filter(elm => elm.director === "Steven Spielberg" && elm.genre.includes("Drama"))
+    return movies.length
+}
+    
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(moviesArray) {
+
+    if (moviesArray.length === 0) {
+        return 0
+    }
+
+    const rateSum = moviesArray.reduce((acc, elm) => {
+
+        if (typeof elm.rate === "number") {
+        
+            return acc + elm.rate
+        }
+        else {
+
```

Comment:
```suggestion
return acc
```
Era en esta iteración donde había que contemplar lo que hablamos ayer, que Jasmine le pasa una película que no tiene rate, y lo manejas bien, pero dejaste la condición del else vacía, y el calculo se rompe.

En este caso, si entraba en el _else_ es que la pelicula no tenía rate, por lo que sumandole 0 la suma no se "rompe", o retornar directamente la media sin sumarle nada, que es lo mismo que sumarle 0.